### ISSUE TITLE
TF1.15 Distribute Mode Error 

### ISSUE THREAD

**Issue by KANGRuipeng** [tensorflow/tensorflow#35384]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos7.6
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): TF1.15
- Python version: python3.6
- CUDA/cuDNN version: 10.0/7.6.2

**Code**

 mirror_strategy = tf.distribute.MirroredStrategy()
 with mirror_strategy.scope():
    model = test_net()
    sgd = tf.keras.optimizers.SGD(lr=0.01, momentum=0.9, decay=1e-4)
    model.compile(
        optimizer=sgd, loss="sparse_categorical_crossentropy", metrics=["accuracy"]
    )

    model.fit(
        x_train, 
        y_train,
        batch_size=256,
        epochs=10,
        validation_data=(x_test, y_test),
        verbose=2,
    )

**The Error Information**

No registered 'MultiDeviceIteratorGetNextFromShard' OpKernel for GPU devices compatible with node {{node MultiDeviceIteratorGetNextFromShard}}
        .  Registered:  device='CPU'

**Comment by KANGRuipeng** [comment]
You have fixed this bug in TF2.0. Is there any update of TF1.15

**Comment by ravikyram** [comment]
@KANGRuipeng 
Looks like code is incomplete.Can you please provide simple standalone code to reproduce the issue in our environment. It helps in localizing the issue faster. Thanks!

**Comment by KANGRuipeng** [comment]
> @KANGRuipeng
> Looks like code is incomplete.Can you please provide simple standalone code to reproduce the issue in our environment. It helps in localizing the issue faster. Thanks!

The test_net is a simple conv-net. You can define it by yourself.

**Comment by KANGRuipeng** [comment]
def block(x):

    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)

    x = tf.keras.layers.GlobalAveragePooling2D()(x)
    x = tf.keras.layers.Flatten()(x)

    x = tf.keras.layers.Dense(10)(x)
    x = tf.keras.layers.Activation(activation=tf.nn.softmax)(x)

    return x


def FCN():  # pylint: disable = invalid-name
    in_tensor = tf.keras.Input(shape=(32, 32, 3))
    out_tensors = block(in_tensor)
    model = tf.keras.Model(inputs=in_tensor, outputs=out_tensors, name="FCN")
    return model

**Comment by KANGRuipeng** [comment]
> def block(x):
> 
> ```
> x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
> x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
> x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
> x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
> 
> x = tf.keras.layers.GlobalAveragePooling2D()(x)
> x = tf.keras.layers.Flatten()(x)
> 
> x = tf.keras.layers.Dense(10)(x)
> x = tf.keras.layers.Activation(activation=tf.nn.softmax)(x)
> 
> return x
> ```
> 
> def FCN(): # pylint: disable = invalid-name
> in_tensor = tf.keras.Input(shape=(32, 32, 3))
> out_tensors = block(in_tensor)
> model = tf.keras.Model(inputs=in_tensor, outputs=out_tensors, name="FCN")
> return model

This is the test_net

**Comment by KANGRuipeng** [comment]
The data we use is CIFAR10

**Comment by anj-s** [comment]
KANGRuipeng @ can you point me to the specific PR/bug that was fixed for TF 2?
Can you add the end to end example(input data included) which reproduces this issue? I am unable to repro it with the example I have. Thanks!

**Comment by KANGRuipeng** [comment]
cifar10 = tf.keras.datasets.cifar10
(x_train, y_train), (x_test, y_test) = cifar10.load_data()
x_train, x_test = (x_train - 128.0) / 128.0, (x_test - 128.0) / 128.0

**Comment by KANGRuipeng** [comment]
> cifar10 = tf.keras.datasets.cifar10
> (x_train, y_train), (x_test, y_test) = cifar10.load_data()
> x_train, x_test = (x_train - 128.0) / 128.0, (x_test - 128.0) / 128.0

this is the input data that I use

**Comment by KANGRuipeng** [comment]
> KANGRuipeng @ can you point me to the specific PR/bug that was fixed for TF 2?
> Can you add the end to end example(input data included) which reproduces this issue? I am unable to repro it with the example I have. Thanks!

any feedback?

**Comment by KANGRuipeng** [comment]
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import sys
import tensorflow as tf

def block(x):
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)
    x =tf.keras.layers.Conv2D(filters=96, kernel_size=3, strides=2, padding="same")(x)

    x = tf.keras.layers.GlobalAveragePooling2D()(x)
    x = tf.keras.layers.Flatten()(x)

    x = tf.keras.layers.Dense(10)(x)
    x = tf.keras.layers.Activation(activation=tf.nn.softmax)(x)

    return x
def FCN():
    in_tensor = tf.keras.Input(shape=(32, 32, 3))
    out_tensors = block(in_tensor)
    model = tf.keras.Model(inputs=in_tensor, outputs=out_tensors, name="FCN")
    return model

**Comment by KANGRuipeng** [comment]
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import os
import numpy as np
import tensorflow as tf

from test_net import FCN

os.environ["CUDA_VISIBLE_DEVICES"] = "1"

# load data and norm to [-1,1]
cifar10 = tf.keras.datasets.cifar10
(x_train, y_train), (x_test, y_test) = cifar10.load_data()
x_train, x_test = (x_train - 128.0) / 128.0, (x_test - 128.0) / 128.0

x_test= x_test[0:9984]
y_test = y_test[0:9984]

# config the used GPU memory size
config = tf.ConfigProto(allow_soft_placement=True)
config.gpu_options.per_process_gpu_memory_fraction = 0.3
tf.keras.backend.set_session(tf.Session(config=config))


mirror_strategy = tf.distribute.MirroredStrategy()
with mirror_strategy.scope():
    model = FCN()
    sgd = tf.keras.optimizers.SGD(lr=0.01, momentum=0.9, decay=1e-4)
    model.compile(
    optimizer=sgd, loss="sparse_categorical_crossentropy", metrics=["accuracy"]
    )

    model.fit(
        x_train, 
        y_train,
        batch_size=256,
        epochs=10,
        validation_data=(x_test, y_test),
        verbose=2,
    )

**Comment by KANGRuipeng** [comment]
@ravikyram @anj-s @tensorflowbutler @ymodak  The up two replies are the net-defination code and the training script.

**Comment by KANGRuipeng** [comment]
Is there any one following this issue?

**Comment by anj-s** [comment]
@KANGRuipeng I am unable to repro this error. If you can share a colab link to the code and error message we can take a look. Also I have seen this message as a warning but not an error so I am surprised that this was a hard error. You can also try moving to TF 2 is that is possible.

**Comment by KANGRuipeng** [comment]
> @KANGRuipeng I am unable to repro this error. If you can share a colab link to the code and error message we can take a look. Also I have seen this message as a warning but not an error so I am surprised that this was a hard error. You can also try moving to TF 2 is that is possible.

Is there any report performance issue of TF1.15 under distributed mode?
